Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -1749,47 +1753,64 @@ public class TimeSlider: UIControl {
         if isObserving {
             removeObservers()
         }
-        map?.addObserver(self, forKeyPath: #keyPath(AGSMap.operationalLayers), options: [.new, .old], context: &TimeSlider.KVOContext)
-        scene?.addObserver(self, forKeyPath: #keyPath(AGSScene.operationalLayers), options: [.new, .old], context: &TimeSlider.KVOContext)
-        geoView?.addObserver(self, forKeyPath: #keyPath(AGSGeoView.timeExtent), options: .new, context: &TimeSlider.KVOContext)
+        
+        // Observe operationalLayers of map
+        mapLayersObservation = map?.observe(\.operationalLayers, options: [.new, .old], changeHandler: { [weak self] (map, change) in
```

Comment:
Do any of these observers update view properties? If so, have you made sure that those updates are always made on the main thread?

Reply:
No, it does not update UI directly. It re-initialize the time slider so, in the end all UI update happens through `layoutSubviews`. Do you thing the code of `layoutSubviews` should be called on main thread?

Reply:
No. `layoutSubviews()` is always called by the system on the main thread, so no changes should be needed there.